scripts/with-signing-key.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+# Run another script with the signing key for the current scope,
+# if set. The environment variable KEY_PASSPHRASE will be set to
+# the key's passphrase, to pass in like so:
+#
+#    echo $KEY_PASSPHRASE | gpg \
+#        ${GPG_PASSPHRASE_FROM_STDIN} \
+#        ...
+#
+# The environment variable $KEY_ID will be set to the ID of the
+# key so you can refer to it from the script.
+set -euo pipefail
+
+if [[ "${1:-}" == "" ]]; then
+    echo "Usage: with-signing-key.sh CMD [ARG...]" >&2
+    echo "">&2
+    echo "Run another command with a preloaded GPG keyring." >&2
+    exit 1
+fi
+
+if [[ "${SIGNING_KEY_SCOPE:-}" == "" ]]; then
+    echo "SIGNING_KEY_SCOPE not set; not signing artifacts." >&2
+    exit 0
```

Comment:
Shouldn't that... still execute the command that was passed? Maybe setting a `SKIP_GPG=1` environment or something?

Reply:
You're right. In your case yes, in the case of the `sign-files.sh` script, doesn't make any sense. How about an `--always` flag?

Reply:
No that doesn't make any sense either because that silently succeeds. Those scripts should just fail. Okay, I will do another environment variable.